components/authn-service/tokens/pg/pg_test.go

```diff
@@ -479,6 +479,69 @@ func TestCreateToken(t *testing.T) {
 	}
 }
 
+func TestPurgeProject(t *testing.T) {
+	ctx := context.Background()
+	store, db := setup(t)
+	assert := assert.New(t)
+
+	cases := map[string]func(*testing.T){
+		"when there are mulitple tokens, some with, some without the project to purge": func(t *testing.T) {
+			description := "Test Token Description"
+			projectToPurge := "projectToPurge"
+
+			id1 := "other_projects"
+			projects1 := []string{"otherproject", "otherproject2"}
+			resp1, err := store.CreateToken(ctx, id1, description, true, projects1)
+			assert.NoError(err, "failed to store token1")
+			assert.ElementsMatch(projects1, resp1.Projects)
+
+			id2 := "unassigned"
+			projects2 := []string{"(unassigned)"}
+			resp2, err := store.CreateToken(ctx, id2, description, true, projects2)
+			assert.NoError(err, "failed to store token2")
+			assert.ElementsMatch(projects2, resp2.Projects)
+
+			id3 := "project1_and_others"
```

Comment:
What is project1--renamed perhaps...?
Same for L510 in this file, plus in teams/conformance_test